pipeline/src/test_util.rs

```diff
@@ -128,12 +149,13 @@ pub fn gen_halo2_proof(file_name: &str, inputs: Vec<Bn254Field>) {
     write_or_panic(vkey_file, |writer| {
         pipeline.export_verification_key(writer).unwrap()
     });
-    let mut pipeline = pipeline.with_vkey_file(Some(vkey_file_path));
 
     // Create the proof before adding the setup and vkey to the backend,
     // so that they're generated during the proof
     let proof: Vec<u8> = pipeline.compute_proof().unwrap().clone();
 
+    let mut pipeline = pipeline.with_vkey_file(Some(vkey_file_path));
```

Comment:
Moved this here to agree with the comment above (run the proof before the vkey is added)